File: fixtures/scale/lua/large/ciphers/byteflip.lua
local function byteflip_encode(text)
  local k = 4
  local parts = {}
  for i = 1, #text, k do
    local chunk = text:sub(i, i + k - 1)
    parts[#parts + 1] = chunk:reverse()
  end
  return table.concat(parts)
end

local function byteflip_decode(text)
  return byteflip_encode(text)
end

return { byteflip_encode = byteflip_encode, byteflip_decode = byteflip_decode }

File: fixtures/scale/javascript/large/src/ciphers/byteflip.js
// Self-inverse codec (byteflip): reverse chunks of 4.
function byteflipEncode(text) {
  const k = 4;
  const parts = [];
  for (let i = 0; i < text.length; i += k) {
    parts.push(text.slice(i, i + k).split('').reverse().join(''));
  }
  return parts.join('');
}

function byteflipDecode(text) {
  return byteflipEncode(text);
}

export default { byteflipEncode, byteflipDecode };
